src/components/Moves.js

```diff
@@ -22,7 +22,35 @@ import {
   Steel,
 } from "../img/typeIcons/images";
 
-const Moves = ({ moveSet }) => {
+const Moves = ({ dexNo }) => {
+  const [moveSet, setMoveSet] = useState([]);
+
+  const getMoveset = async (movesArray, n) => {
+    const nMoves = [];
+    for (let i = 0; i < n; i += 1) {
+      const moveUrl = movesArray[i].move.url;
+      try {
+        const moveData = await fetchMove(moveUrl);
+        if (moveData) nMoves.push(moveData);
+      } catch (error) {
+        console.error(error);
+      }
+    }
```

Comment:
Couldn't you use `Promise.all` to fetch all the moves URL?

Reply:
The thing with `Promise.all` is that if even one promise rejects, it will automatically reject.  This is a bit safer.

Reply:
Makes sense